tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -379,13 +378,20 @@ connect_redirect_test(
 
         sender_socket->get_received_message(bytes_received, received_message);
 
-        // Remote redirection expects the SERVER_MESSAGE response.
-        // Local redirection expects the redirect context response.
+        // For local redirection, the redirect context is expected to be set and returned.
+        // If the connection is not redirected or is redirected to a remote address,
+        // check for the SERVER_MESSAGE generic response.
+        bool redirected =
+            (destination_port != proxy_port || !INETADDR_ISEQUAL((SOCKADDR*)&destination, (SOCKADDR*)&proxy));
+        bool local_redirect = !(
+            INETADDR_ISEQUAL((SOCKADDR*)&proxy, (SOCKADDR*)&_globals.addresses[socket_family_t::IPv4].remote_address) ||
+            INETADDR_ISEQUAL((SOCKADDR*)&proxy, (SOCKADDR*)&_globals.addresses[socket_family_t::IPv6].remote_address) ||
+            INETADDR_ISEQUAL((SOCKADDR*)&proxy, (SOCKADDR*)&_globals.addresses[socket_family_t::Dual].remote_address));
```

Comment:
qq -- would this not return true for _all_ redirections? (both local and remote)?

Reply:
Can you double check my read here - 'proxy' will either be loopback, local, or remote address. If it is loopback or local, it should evaluate to !(false || false || false), which would be true. If the 'proxy' is the remote address, then it should evaluate to !(true || false || false) [in some order], which would be false.

Reply:
You are right. I (wrongly) though the original destination is always saved in `remote_address`.